stdlib/src/builtin/builtin_list.mojo

```diff
@@ -290,6 +290,12 @@ struct _lit_mut_cast[
     ]
 
 
+struct _is_mutable_origin[
```

Comment:
**Question** Can you help me understand why this is needed?  You should be able to directly access the `is_mutable` from the call sites without any issues.

Reply:
Not on type signatures where all you have is `__origin_of()` and you want to extract the mutability of that:
```mojo
    @staticmethod
    @always_inline("nodebug")
    fn address_of(
        ref [_, address_space._value.value]arg: type
    ) -> UnsafePointer[
        type,
        address_space,
        is_mutable = _is_mutable_origin[__origin_of(arg)].result,
        origin = __origin_of(arg),
    ] as result:
        """Gets the address of the argument.

        Args:
            arg: The value to get the address of.

        Returns:
            An UnsafePointer which contains the address of the argument.
        """
        return __type_of(result)(
            __mlir_op.`lit.ref.to_pointer`(__get_mvalue_as_litref(arg))
        )
```
Though if there is a cleaner way to extract an attribute from `__origin_of()` I'm all ears